src/test/scheduler_tests.cpp

```diff
@@ -155,4 +155,46 @@ BOOST_AUTO_TEST_CASE(singlethreadedscheduler_ordered)
     BOOST_CHECK_EQUAL(counter2, 100);
 }
 
+BOOST_AUTO_TEST_CASE(mockforward)
+{
+    CScheduler scheduler;
+
+    int counter{0};
+    CScheduler::Function dummy = [&counter](){counter++;};
+
+    // schedule jobs for 2, 5 & 8 minutes into the future
+    int64_t min_in_milli = 60*1000;
+    scheduler.scheduleFromNow(dummy, 2*min_in_milli);
+    scheduler.scheduleFromNow(dummy, 5*min_in_milli);
+    scheduler.scheduleFromNow(dummy, 8*min_in_milli);
+
+    // check taskQueue
+    boost::chrono::system_clock::time_point first, last;
+    size_t num_tasks = scheduler.getQueueInfo(first, last);
+    BOOST_CHECK(num_tasks == 3);
+
+    boost::thread scheduler_thread(std::bind(&CScheduler::serviceQueue, &scheduler));
+
+    // bump the scheduler forward 5 minutes
+    scheduler.MockForward(boost::chrono::seconds(5*60));
+
+    // finish up
+    MicroSleep(600);
+    scheduler.stop(false);
```

Comment:
ca57312e5b0dc9d28766369b963395f1690e0b4f

Maybe change this to:
```
scheduler.scheduleFromNow([&]() { scheduler.stop(false); }, 1);
```
and remove the sleep above.

Reply:
clever. thank you! 

lets see if travis likes it 🤞🏽

Reply:
yay! looking good. only travis failure seemed unrelated.